contracts/pair/src/utils.rs

```diff
@@ -24,6 +25,60 @@ pub fn assert_approx_ratio(a: Decimal, b: Decimal, tolerance: Decimal) -> bool {
     diff <= tolerance
 }
 
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct TokenInitInfo {
+    pub token_wasm_hash: BytesN<32>,
+    pub token_a: Address,
+    pub token_b: Address,
+}
+
+impl TokenInitInfo {
+    pub fn get_token_wasm_hash(&self) -> BytesN<32> {
+        self.token_wasm_hash.clone()
+    }
+
+    pub fn get_token_a(&self) -> Address {
+        self.token_a.clone()
+    }
+
+    pub fn get_token_b(&self) -> Address {
+        self.token_b.clone()
+    }
+}
```

Comment:
What's the purpose of those helpers?
1) You already made all parameters `pub` within `TokenInitInfo`
2) This only increases binary's size.

Just access each parameter directly in code.